spec/services/proforma_service/convert_task_to_exercise_spec.rb

```diff
@@ -38,32 +38,54 @@
         uuid: 'uuid',
         parent_uuid: 'parent_uuid',
         language: 'language',
-        meta_data: {
-          CodeOcean: {
-            instructions: 'instructions',
-          },
-        },
+        meta_data: meta_data,
         model_solutions: model_solutions,
         files: files,
         tests: tests
       )
     end
     let(:user) { create(:teacher) }
+
     let(:files) { [] }
     let(:tests) { [] }
     let(:model_solutions) { [] }
     let(:exercise) { nil }
 
+    let(:meta_data) do
+      {
+        CodeOcean: {
+          public: public,
+          hide_file_tree: hide_file_tree,
+          allow_file_creation: allow_file_creation,
+          allow_auto_completion: allow_auto_completion,
+          expected_difficulty: expected_difficulty,
+          execution_environment_id: execution_environment.id,
+          files: files_meta_data,
+        },
+      }
+    end
+    let(:public) { 'true' }
+    let(:hide_file_tree) { 'true' }
+    let(:allow_file_creation) { 'true' }
+    let(:allow_auto_completion) { 'true' }
+    let(:expected_difficulty) { 7 }
+    let!(:execution_environment) { create(:java) }
+    let(:files_meta_data) { {} }
```

Comment:
Is this something we should extract to a factory? We are mainly passing static variables, aren't we?

Reply:
In a previous test, we ensure to update the metadata. Therefore this test doesn’t need to verify it.